Fix location tracking notification

seadowg opened the pull request:
I noticed when testing out the new geopoint widget that the "Tracking location..." notification looks wrong:

<img width="302" alt="Screenshot 2022-01-23 at 16 11 18" src="https://user-images.githubusercontent.com/556280/150691365-6127af18-1d6b-44b3-a10e-606106d3b68e.png">

Turns out that it hadn't been configured properly - the notification icon was `null`. I've fixed it:

<img width="302" alt="Screenshot 2022-01-23 at 17 53 11" src="https://user-images.githubusercontent.com/556280/150691388-227bb43e-0355-440e-b865-bda1ea5053dd.png">

#### What has been done to verify that this works as intended?

Verified manually.

#### Why is this the best possible solution? Were any other approaches considered?

Comments inline.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Should just fix the notification. Not much risk to anything else here.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

seadowg committed:
Pull out helper for static value

seadowg commented on collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.kt:
```diff
@@ -31,10 +28,7 @@ class ApplicationInitializer(
-    private val analyticsInitializer: AnalyticsInitializer,
-    private val projectsRepository: ProjectsRepository,
-    private val settingsProvider: SettingsProvider
+    private val initializers: List<Initializer>
```
We've talked before about not wanting to use the Jetpack App Initialization framework and just continue to develop our own custom approach. I wanted to move to using a list of initializers here rather than passing another explicit one in - this made it easier to add the new initializer and it'll make it easy to add others in the future.

seadowg commented on collect_app/src/main/java/org/odk/collect/android/injection/config/AppDependencyModule.java:
```diff
@@ -584,7 +587,17 @@ public UpgradeInitializer providesUpgradeInitializer(Context context, SettingsPr
     @Provides
     public ApplicationInitializer providesApplicationInitializer(Application context, UserAgentProvider userAgentProvider, PropertyManager propertyManager, Analytics analytics, UpgradeInitializer upgradeInitializer, AnalyticsInitializer analyticsInitializer, ProjectsRepository projectsRepository, SettingsProvider settingsProvider) {
-        return new ApplicationInitializer(context, userAgentProvider, propertyManager, analytics, upgradeInitializer, analyticsInitializer, projectsRepository, settingsProvider);
+        return new ApplicationInitializer(context, userAgentProvider, propertyManager, analytics, asList(
```
We might want to introduce dependencies for `Initializer`s in the future but at the moment it's nice to be able to see the order they are executed in to make sure any interdependencies between them work out ok.

seadowg commented on collect_app/src/main/java/org/odk/collect/android/utilities/IconUtils.java:
```diff
@@ -31,10 +31,6 @@
     private IconUtils() {
     }
 
-    public static int getNotificationAppIcon() {
```
This didn't feel worth hanging on to

grzesiek2010 commented on collect_app/src/main/java/org/odk/collect/android/application/initialization/LocationInitializer.kt:
```diff
@@ -0,0 +1,14 @@
+class LocationInitializer : Initializer {
+
+    override fun initialize() {
+        ForegroundServiceLocationTracker.notificationIcon = R.drawable.ic_notes_white
```
Can't we just use this icon statically in `ForegroundServiceLocationTracker` like we do in case of all the other notifications? Do we need to initialize it like that?

seadowg replied:
I guess my thinking was that the icon is "Collect" specific and shouldn't really be in the `location` module

grzesiek2010 replied:
I see then that icon could be in `androidshared` or even we could have a separate module for drawables like we have for strings as it is possible to use the same drawables in different modules (rather in the future). You are a master of abstractions and I usually like the way you think but here it's too much for me I guess especially now as you want to add this to v2022.1 it should be a small fix without touching all those initializes (you can do that in a separate pr).

seadowg replied:
> You are a master of abstractions and I usually like the way you think

🙏

> ...but here it's too much for me I guess especially now as you want to add this to v2022.1

Yeah I do see what you're saying. I'm trying to adhere to a self enforced rule that we should be able to pull the module out and use it in another application but needing to do initialization like this is a pretty clunky way of making that work. We can probably do something nicer with a shared interface for grabbing the "notification icon" from the `Application` but for now let's just move the icon to `location`. We're using a different icon for `AudioRecorderService` which I don't think makes sense - really all the app notifications should use a consistent icon. I can create an issue for harmonizing that and we can think about a more "elegant" way to configure the icon then.

grzesiek2010 replied:
Ok

seadowg committed:
Move notification icon to location module

We should work out a way to share this between modules in the future
but right now everything will compile it it lives here.

seadowg committed:
Rename icon

kkrawczyk123 commented:
**Tested with success!**
Verified on Androids: 5.1 and 8.1

Verified cases:
- geopoint with no appearance 
- regression check on other notifications: autosend in setting/forms, form update download, autodownload

srujner commented:
**Tested with success!**
Verified on Androids: 10 and 11
